src/components/timeline/VerticalIndicators.js

```diff
@@ -26,13 +26,21 @@ type Props = {|
   +rangeEnd: Milliseconds,
   +zeroAt: Milliseconds,
   +width: CssPixels,
+  +shouldShowTooltip: boolean,
+  +onRightClick: MarkerIndex => mixed,
 |};
 
 /**
  * This component draws vertical indicators from navigation related markers for a track
  * in the timeline.
  */
 export class VerticalIndicators extends React.PureComponent<Props> {
+  _onMouseDown = (e: SyntheticMouseEvent<HTMLDivElement>) => {
+    if (e.button === 2 && this.props.onRightClick && e.currentTarget) {
+      this.props.onRightClick(parseInt(e.currentTarget.dataset.markerIndex));
```

Comment:
This wasn't so easy to make all this work properly. I decided to gather all the state at the parent component to make it easier.